src/handlers/expiry/providers/DateProvider.ts

```diff
@@ -183,11 +183,20 @@ export class DateProvider {
   public static addOneYearMinusOneDayISOString(
     inputDate: Date | string
   ): string {
-    return moment(inputDate)
-      .add(1, "year")
-      .subtract(1, "day")
-      .startOf("day")
-      .toISOString();
+
+    const generateExpiryDate = (day: number = 1): string => {
```

Comment:
Since a leap year is your edge case, it probably shouldn't be the default on a function called generateExpiryDate, it should be a passed parameter, this would effect the return statement below. You could also change the parameter name from day -> subtractDays

Reply:
Fair point, i'll make these changes when the story is pulled back into sprints. It's been 'on-hold' since Nov 2021, but will implement both suggestions when it's brought back in scope.

Reply:
It's been a while since I have looked at this code, but the default is actually to deduct 1 day, which is what happens when it is not a leap year but I take your point on passing in the parameter and can see where the confusion could happen.

Reply:
Cool, that makes sense now 😄